### PR TITLE
feature/05-create-custom-input

### PR DESCRIPTION
**Pull request by zohaibwebdev**

### CODE REVIEW

File: src/shared/types/input.types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export interface InputPropsTypes {
```
Comment (awaisshahid965) [inline src/shared/types/input.types.ts]: These are not going to be used project wise, move it with component

File: src/shared/components/custom-input/input.component.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { FC } from 'react'
+import { InputPropsTypes } from '@/shared/types/input.types'
+
+const CustomInput: FC<InputPropsTypes> = ({
```
Comment (awaisshahid965) [inline src/shared/components/custom-input/input.component.tsx]: Why your component name and file name differs? Both needs to be same

File: src/shared/components/custom-input/input.component.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            {label && <label>{label}:</label>}
+            <input
+                name={name}
+                type={type}
+                defaultValue={defaultValue}
+                placeholder={placeholder}
+                required={required}
+                onChange={handleChange}
+                className={className}
+                readOnly={readonly}
```
Comment (awaisshahid965) [inline src/shared/components/custom-input/input.component.tsx]: We need a prop to add default value

File: src/shared/components/custom-input/input.component.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    defaultValue,
+    className,
+    changeEvent,
+    readonly,
+}) => {
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        changeEvent?.(e)
+    }
+    return (
+        <div>
```
Comment (awaisshahid965) [inline src/shared/components/custom-input/input.component.tsx]: No styling for this???

File: src/shared/components/custom-input/input.component.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+const CustomInput: FC<InputPropsTypes> = ({
+    label,
+    type,
+    placeholder,
+    required,
+    errorMessage,
+    name,
+    defaultValue,
+    className,
```
Comment (awaisshahid965) [inline src/shared/components/custom-input/input.component.tsx]: Only accepting classes for input, what if we need to make some changes to label or their parent from outside. You should keep Open-Close principle in mind while creating reusable components

**Review by awaisshahid965 — CHANGES_REQUESTED** [review]

File: src/shared/types/input.types.ts (hunk 1 above)
Comment (zohaibwebdev) [reply src/shared/types/input.types.ts]: done it by initilizing it in the component

File: src/shared/components/custom-input/input.component.tsx (hunk 2 above)
Comment (zohaibwebdev) [reply src/shared/components/custom-input/input.component.tsx]: ok

File: src/shared/components/custom-input/input.component.tsx (hunk 4 above)
Comment (zohaibwebdev) [reply src/shared/components/custom-input/input.component.tsx]: done it by making flex container and flex direction is colum

File: src/shared/components/custom-input/input.component.tsx (hunk 5 above)
Comment (zohaibwebdev) [reply src/shared/components/custom-input/input.component.tsx]: yes now label classes available

**Review by awaisshahid965 — APPROVED** [review]

**Pull request merged**